x/stakeibc/keeper/msg_server_fast_unbond.go

```diff
@@ -0,0 +1,117 @@
+package keeper
+
+import (
+	"context"
+	"fmt"
+	epochtypes "github.com/Stride-Labs/stride/v5/x/epochs/types"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+
+	"github.com/Stride-Labs/stride/v5/x/stakeibc/types"
+)
+
+func (k msgServer) FastUnbond(goCtx context.Context, msg *types.MsgFastUnbond) (*types.MsgFastUnbondResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+	k.Logger(ctx).Info(fmt.Sprintf("fast unbond: %s", msg.String()))
+
+	// get our addresses, make sure they're valid
+	sender, err := sdk.AccAddressFromBech32(msg.Creator)
+	if err != nil {
+		return nil, sdkerrors.Wrapf(sdkerrors.ErrInvalidAddress, "creator address is invalid: %s. err: %s", msg.Creator, err.Error())
+	}
+
+	// then make sure host zone is valid
+	hostZone, found := k.GetHostZone(ctx, msg.HostZone)
+	if !found {
+		return nil, sdkerrors.Wrapf(types.ErrInvalidHostZone, "host zone is invalid: %s", msg.HostZone)
+	}
+
+	// get the coins to return, they need to be in the format {amount}{denom}
+	// is safe. The converse is not true.
+	ibcDenom := hostZone.GetIbcDenom()
+	coinString := msg.Amount.String() + ibcDenom
+	outCoin, err := sdk.ParseCoinNormalized(coinString)
```

Comment:
I'm finding this naming schema confusing with
outCoin
nativeAmount
inCoin

How can we rename this to make this more clear?

Reply:
OK, I think I have fixed this.  I found the previous schema confusing as well.  Let me know what you think.